src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5120,7 +5120,7 @@ def list_generator(pages, num_results=50):
 
         # handle num results
         if num_results is not None:
-            if num_results == len(result):
+            if num_results >= len(result):
```

Comment:
This is an unrelated code detail optimization, because there are few changes to it.